### ISSUE TITLE
Gonex stops on fastsync mode

### ISSUE THREAD

**Issue by dapprocket** [nextyio/gonex#214]
#### System information

Gonex version: gonex/v2.0.0-stable
OS & Version: Ubuntu 18.04

Gonex stops on fastsync after syncing for period after an unfixed block number.

Ran with --verbosity=9, it was still traced but gonex.log returned nothing more

The last gathered logs:

INFO [06-03|21:48:50.999] Regenerated local transaction journal    transactions=0 accounts=0
INFO [06-03|22:48:50.999] Regenerated local transaction journal    transactions=0 accounts=0
INFO [06-03|23:48:50.999] Regenerated local transaction journal    transactions=0 accounts=0

**Comment by hadv** [comment]
it seems problem of go-routine and channel then it's non-trivial to figure out the root cause. After -sync to a block number then node stop insert block header and report as completed

```
DEBUG[06-04|04:33:00.763] Fetching batch of headers                id=daf0b33dfe55b0c9 conn=staticdial count=192 fromnum=319297 skip=0   reverse=false
DEBUG[06-04|04:33:01.106] Data fetching completed                  type=headers
DEBUG[06-04|04:33:01.106] Skeleton fill terminated                 err=nil
```

**Comment by Zergity** [comment]
Don't worry about this now, I can take care of this next month.

**Comment by hadv** [comment]
> Don't worry about this now, I can take care of this next month.

Sure, just make some quick investigate to see what is the problem but it seem not easy one

**Comment by hadv** [comment]
From the ThangLong hardfork, node need to get data from state to verify header. But when run node in `--fast` sync mode, the state is not available when fetching headers data that make the verify header failed and node cannot proceed sync anymore. 

We're discussing on #237 to finalize a concrete solution to support `--fast` sync mode.

**Comment by Zergity** [comment]
Will be solved by #238
